File: tests/fixtures/csharp-lexical-syntax/host/src/CSharpCohort/Syntax.cs
namespace CSharpCohort;

public static class Syntax
{
    public static Invoice HandledParse(string raw)
    {
        if (!string.IsNullOrWhiteSpace(raw))
        {
            return BillingParser.ParseBilling(raw);
        }

        return new Invoice("empty", 0);
    }

    public static Invoice UnhandledParse(string raw) => BillingParser.ParseBilling(raw);

    public static int RouteInvoice(int value)
    {
        var score = 0;
        if (value > 0) score++;
        if (value > 1) score++;
        if (value > 2) score++;
        if (value > 3) score++;
        if (value > 4) score++;
        if (value > 5) score++;
        if (value > 6) score++;
        if (value > 7) score++;
        return score;
    }
}

File: tests/fixtures/kotlin-lexical-syntax/host/src/main/kotlin/cohort/Syntax.kt
package cohort

fun guardedParse(raw: String): Invoice {
    if (raw.isNotBlank()) {
        return parseBilling(raw)
    }
    return Invoice("missing", 0)
}

fun unhandledParse(raw: String): Invoice = parseBilling(raw)

fun routeInvoice(invoice: Invoice, flags: List<Boolean>): Int {
    var score = 0
    if (invoice.amount > 0) score += 1
    if (invoice.id.isNotBlank()) score += 1
    for (flag in flags) if (flag) score += 1
    while (score < 2) score += 1
    when (invoice.amount) {
        0 -> score += 1
        else -> score += 2
    }
    if (flags.isNotEmpty() && flags.first()) score += 1
    if (invoice.amount > 1 || invoice.id.length > 2) score += 1
    return score
}
